sdk/src/impl/position-impl.ts

```diff
@@ -189,6 +190,75 @@ export class PositionImpl implements Position {
     return txBuilder;
   }
 
+  async collectFees(
+    resolveATA: boolean = true,
+    destinationWallet?: Address,
+    positionWallet?: Address,
+    ataPayer?: Address
+  ): Promise<TransactionBuilder> {
+    const [destinationWalletKey, positionWalletKey, ataPayerKey] = AddressUtil.toPubKeys([
+      destinationWallet ?? this.ctx.wallet.publicKey,
+      positionWallet ?? this.ctx.wallet.publicKey,
+      ataPayer ?? this.ctx.wallet.publicKey,
+    ]);
+
+    const whirlpool = await this.fetcher.getPool(this.data.whirlpool, true);
+    if (!whirlpool) {
+      throw new Error("Unable to fetch whirlpool for this position.");
+    }
+
+    // Question @meep: Can either fee_owed_a or fee_owed_b be 0 while the other is non-zero?
+    //                 If so, will the collect_fees IX fail since it just blindly calls a token transfer CPI?
```

Comment:
collectfee/rewards won't fail if there's no fee to collect. It'll just be a no-op.

> Can either fee_owed_a or fee_owed_b be 0 while the other is non-zero?

Yes! Fee comes from the swap-input-token, so if a pool is newly created and it only got swapped in 1 direction, you would end up with that scenario. But as explained above, it's not something the client would have to explicitly handle.